src/engine.cc

```diff
@@ -236,31 +220,8 @@ void EngineController::UpdateFromUciOptions() {
   }
 
   // Network.
-  std::string network_path = options_.Get<std::string>(kWeightsId.GetId());
-  std::string backend = options_.Get<std::string>(kNnBackendId.GetId());
-  std::string backend_options =
-      options_.Get<std::string>(kNnBackendOptionsId.GetId());
-
-  if (network_path == network_path_ && backend == backend_ &&
-      backend_options == backend_options_)
-    return;
-
-  network_path_ = network_path;
-  backend_ = backend;
-  backend_options_ = backend_options;
-
-  std::string net_path = network_path;
-  if (net_path == kAutoDiscover) {
-    net_path = DiscoverWeightsFile();
-  } else {
-    std::cerr << "Loading weights file from: " << net_path << std::endl;
-  }
-  Weights weights = LoadWeightsFromFile(net_path);
-
-  OptionsDict network_options(&options_);
-  network_options.AddSubdictFromString(backend_options);
-
-  network_ = NetworkFactory::Get()->Create(backend, weights, network_options);
+  std::unique_ptr<Network> tmp_net = NetworkFactory::LoadNetwork(options_);
+  if (tmp_net) network_.swap(tmp_net);
```

Comment:
Why is that? I think if network failed to load, it's better to do something than ignore that. E.g. throw an exception?

Also with unique_ptrs, it's more idiomatic to write `network_ = std::move(tmp_net);` or directly `network_ = NetworkFactory::LoadNetwork(options_);`

Reply:
Here, nullptr is not an indication of failure to load but that we are reusing the already loaded network, so it souldn't change. This was the cleaner way I found to refactor the load logic from engine.cc and tournament.cc. I'll do the change to move(), it has to be conditional.